tests/foreman/api/test_hostcollection.py

```diff
@@ -379,21 +419,64 @@ def test_negative_create_with_invalid_name(self):
                     entities.HostCollection(name=name, organization=self.org).create()
 
     @tier1
-    @stubbed()
     def test_positive_add_subscription(self):
-        """Try to add a subscription to a host collection
+        """Try to bulk add a subscription to members of a host collection.
 
         :id: c4ec5727-eb25-452e-a91f-87cafb16666b
 
         :steps:
 
             1. Create a new or use an existing subscription
-            2. Add the subscription to the host collection
+            2. Add the subscription to the members of the host collection
 
-        :expectedresults: The subscription was added to the host collection
+        :expectedresults: The subscription was added to hosts in the host collection
 
         :CaseImportance: Critical
         """
+        # Create an activate key for this test
+        act_key = entities.ActivationKey(
+            content_view=self.content_view, environment=self.lce, organization=self.org
+        ).create()
+        # this command creates a host collection and "appends", makes available, to the AK
+        act_key.host_collection.append(entities.HostCollection(organization=self.org).create())
+        # Move HC from Add tab to List tab on AK view
+        act_key = act_key.update(['host_collection'])
+        # Create a product to be bulk added to members of Host Collection
+        product = entities.Product(organization=self.org).create()
+        prod_name = product.name
+        product_subscription = entities.Subscription().search(
+            query={'search': f'name={prod_name}'}
+        )[0]
+        with VirtualMachine(distro=DISTRO_RHEL7) as client1, VirtualMachine(
+            distro=DISTRO_RHEL7
+        ) as client2:
+            clients = [client1, client2]
+            for client in clients:
+                client.install_katello_ca()
+                client.register_contenthost(self.org.label, act_key.name)
+            # Read host_collection back from Satellite to get host_ids
+            host_collection = act_key.host_collection[0].read()
+            host_ids = [host.id for host in host_collection.host]
+            # Call nailgun to make the API PUT to members of Host Collection
+            entities.Host().bulk_add_subscriptions(
+                data={
+                    "organization_id": self.org.id,
+                    "included": {"ids": host_ids},
+                    "subscriptions": [{"id": product_subscription.id, "quantity": 1}],
+                }
+            )
+            # Get the subscriptions from hosts
+            for host_id in host_ids:
+                req = request(
+                    'GET',
+                    '{0}/api/v2/hosts/{1}/subscriptions'.format(
+                        settings.server.get_url(), host_id
+                    ),
```

Comment:
Don't we have nailgun support for this endpoint?
If we do, we should use it.
If not, we should add it. I probably wouldn't block this PR until support is added, but I would recommend adding it as soon as possible and switching this part over to new nailgun method.

Reply:
Support for that comes from
`from nailgun.client import request`
 
I did search in nailgun repo, it seems I am the first to use this call. I can raise an issue to keep track of such cases.